internal/api/shared/database/sql/query.go

```diff
@@ -0,0 +1,124 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databaseSQL
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/huandu/go-sqlbuilder"
+	"github.com/perses/perses/internal/api/interface/v1/dashboard"
+	"github.com/perses/perses/internal/api/interface/v1/datasource"
+	"github.com/perses/perses/internal/api/interface/v1/folder"
+	"github.com/perses/perses/internal/api/interface/v1/globaldatasource"
+	"github.com/perses/perses/internal/api/interface/v1/project"
+	databaseModel "github.com/perses/perses/internal/api/shared/database/model"
+	modelAPI "github.com/perses/perses/pkg/model/api"
+	modelV1 "github.com/perses/perses/pkg/model/api/v1"
+)
+
+func generateProjectResourceInsertQuery(tableName string, id string, rowJSONDoc []byte, metadata *modelV1.ProjectMetadata) (string, []interface{}) {
+	return sqlbuilder.NewInsertBuilder().
+		InsertInto(tableName).
+		Cols(colID, colName, colProject, colDoc).
+		Values(id, metadata.Name, metadata.Project, rowJSONDoc).
+		Build()
+}
+
+func generateResourceInsertQuery(tableName string, id string, rowJSONDoc []byte, metadata *modelV1.Metadata) (string, []interface{}) {
+	return sqlbuilder.NewInsertBuilder().
+		InsertInto(tableName).
+		Cols(colID, colName, colDoc).
+		Values(id, metadata.Name, rowJSONDoc).
+		Build()
+}
+
+func (d *DAO) generateInsertQuery(entity modelAPI.Entity) (string, []interface{}, error) {
+	id, tableName, idErr := d.getIDAndTableName(modelV1.Kind(entity.GetKind()), entity.GetMetadata())
+	if idErr != nil {
+		return "", nil, idErr
+	}
+	rowJSONDoc, unmarshalErr := json.Marshal(entity)
+	if unmarshalErr != nil {
+		return "", nil, unmarshalErr
+	}
+	var sql string
+	var args []interface{}
+	switch m := entity.GetMetadata().(type) {
+	case *modelV1.ProjectMetadata:
+		sql, args = generateProjectResourceInsertQuery(tableName, id, rowJSONDoc, m)
+	case *modelV1.Metadata:
+		sql, args = generateResourceInsertQuery(tableName, id, rowJSONDoc, m)
+	}
+	return sql, args, nil
+}
+
+func (d *DAO) generateUpdateQuery(entity modelAPI.Entity) (string, []interface{}, error) {
+	id, tableName, idErr := d.getIDAndTableName(modelV1.Kind(entity.GetKind()), entity.GetMetadata())
+	if idErr != nil {
+		return "", nil, idErr
+	}
+	rowJSONDoc, unmarshalErr := json.Marshal(entity)
+	if unmarshalErr != nil {
+		return "", nil, unmarshalErr
+	}
+	builder := sqlbuilder.NewUpdateBuilder().Update(tableName)
+	builder.Where(builder.Equal(colID, id))
+	builder.Set(builder.Assign(colDoc, rowJSONDoc))
+	sql, args := builder.Build()
+	return sql, args, nil
+}
+
+func generateProjectResourceSelectQuery(tableName string, project string, name string) (string, []interface{}) {
+	queryBuilder := sqlbuilder.NewSelectBuilder().
+		Select(colDoc).
+		From(tableName)
+	if len(name) > 0 {
+		queryBuilder.Where(queryBuilder.Like(colName, fmt.Sprintf("%s%%", name)))
```

Comment:
Maybe I'm missing context here but shouldn't it be `%%s%` instead ? to generate something like `WHERE col LIKE '%My Dashboard%'`, otherwise here it would give `WHERE col LIKE 'My Dashboard%%'`